components/script/dom/webgltexture.rs

```diff
@@ -416,6 +426,114 @@ impl WebGLTexture {
     pub fn detach_from_framebuffer(&self) {
         self.attached_framebuffer.set(None);
     }
+
+    pub fn storage(
+        &self,
+        target: u32,
+        levels: i32,
+        internal_format: u32,
+        width: i32,
+        height: i32,
+        depth: i32,
+    ) -> WebGLResult<()> {
+        if self.is_immutable.get() {
+            return Err(WebGLError::InvalidOperation);
+        }
+
```

Comment:
I believe we should call `self.initialize(..)` here like https://github.com/servo/servo/blob/2c77ab66d4e718437a236d8e48082ab670ea254f/components/script/dom/webglrenderingcontext.rs#L733-L745 . This is where the missing texture info comes from.